### ISSUE TITLE
RoleAssignmentScheduleRequest and RoleEligibilityScheduleRequest are missing in Microsoft.Authorization

### ISSUE THREAD

**Issue by NexiStein** [pulumi/pulumi-azure-native#2073]
### What happened?

I have tried to import numerous versions of Microsoft.Authorization (V20221001, V20220401) in c#, but they all seem to missing several endpoints which do exist in the [API](https://learn.microsoft.com/en-us/rest/api/authorization/). Two examples are:

https://learn.microsoft.com/en-us/rest/api/authorization/role-assignment-schedule-requests/get?tabs=HTTP
https://learn.microsoft.com/en-us/rest/api/authorization/role-eligibility-schedule-requests/get?tabs=HTTP

### Steps to reproduce
I tested using a very simple azure-csharp setup:
```
using Pulumi.AzureNative.Authorization.V20220401;
return await Pulumi.Deployment.RunAsync(() =>
{
    // Unable to find the endpoints here
});
```
### Expected Behavior

I expected the Azure management API enpdoints to exist in Pulumi.

### Actual Behavior

The endpoints were not there :-)

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by squaremo** [comment]
I can confirm I see those represented in the API definitions (in this repo, they get downloaded to `azure-rest-api-specs` during the build, and the file for the first missing API is azure-rest-api-specs/specification/authorization/resource-manager/Microsoft.Authorization/stable/2020-10-01/RoleAssignmentScheduleRequest.json); but it doesn't end up represented in the generated schema or the SDKs.

After putting in a few printlns I can see that the first API endpoint in question doesn't have the right "shape" to be considered a Pulumi resource; specifically, it fails at this line: https://github.com/pulumi/pulumi-azure-native/blob/master/provider/pkg/openapi/discover.go#L297 (with neither `hasDelete` nor `hasDefault` true). I suspect something similar is also the case with the other endpoint.

While this aren't mechanically converted into a part of the SDK, perhaps it's possible to add them in with hand-written code. I'll have to defer to @danielrbradley et al on that question!

**Comment by danielrbradley** [comment]
Where an API only provides PUT and GET endpoints (no DELETE), we need to know what the "default value" of the resource should be. 

## Implementation notes

The default value should be added to https://github.com/pulumi/pulumi-azure-native/blob/master/provider/pkg/openapi/defaultResourcesState.go

Before we can implemented this we need to research what would be appropriate default values to reset back to on delete.

**Comment by aucampia** [comment]
I suspect `POST .../cancel` [[ref](https://learn.microsoft.com/en-us/rest/api/authorization/role-assignment-schedule-requests/get?view=rest-authorization-2020-10-01&tabs=HTTP),[ref](https://learn.microsoft.com/en-us/rest/api/authorization/role-eligibility-schedule-requests/get?view=rest-authorization-2020-10-01&tabs=HTTP)] should be used instead of the verb `DELETE`. Maybe the conversion rules can be changed to accommodate `POST .../cancel`?

**Comment by aucampia** [comment]
Just for reference sake, these resources are broken in Bicep for what I suspect similar reasons: https://github.com/Azure/bicep-types-az/issues/1956

**Comment by mikhailshilkov** [comment]
Duplicate of https://github.com/pulumi/pulumi-azure-native/issues/2455, let's track there since it has a more crisp description
